app/src/main/java/com/duckduckgo/app/browser/BrowserActivity.kt

```diff
@@ -260,6 +260,20 @@ class BrowserActivity : DuckDuckGoActivity(), BookmarkDialogCreationListener {
                 }
 
         omnibarTextInput.addTextChangedListener(object : TextChangedWatcher() {
+
+            override fun onTextChanged(
+                charSequence: CharSequence,
+                start: Int,
+                before: Int,
+                count: Int
+            ) {
+                // some 3rd party keyboards submit \n instead of an IME action
+                if (before == 0 && count == 1 && charSequence[start] == '\n') {
```

Comment:
Do we still need this is we get rid of the "shortMessage" ime option in the xml

Reply:
I guess we're still at the mercy of the 3rd party keyboards... even if changing this changes this one individual keyboard's behaviour, now that we've seen it once i'm tempted to leave it in as a safety.

Reply:
Keyboard behavior is complex on android and I'm not keen to introduce patches unless we know they fix a legitimate issue lest we introduce something knew. For example I often share text from other applications to our app to search for stuff. If there's a line break in it this patch (that we're not sure we need) may break that.

As we were causing this by using an incorrect IME option, I'd rather remove that alone.

Reply:
If you can find evidence that we need this, I'll happily approve it.

Reply:
>As we were causing this by using an incorrect IME option
I disagree we are to blame, or that this is an incorrect IME option. It seems like an entirely appropriate IME option to use here (https://developer.android.com/reference/android/text/InputType.html#TYPE_TEXT_VARIATION_SHORT_MESSAGE)

That said, I'm happy to remove `shortMessage`, and the patch and wait and see if there are any other reports.

>Keyboard behavior is complex
Ain't it just!

Reply:
The reason I feel it's wrong is because it introduces a bunch of behavior that just doesn't work with urls. For example, I think it's the cause of other keyboard issues we've seen around extra spacing being added when typing. This isn't our fault - google should document the behavior of these options better and be clear about which combinations work well together and which ones fight each other.

Reply:
> Ain't it just!
Yesssss! Damn you android keyboards. I get the feeling we'll get to know them all well. The change we've deferred here may yet be revived 😆

Reply:
Like I would! 😏